### ISSUE TITLE
Test suite breaks test discovery if custom User is installed.

### ISSUE THREAD

**Issue by spookylukey** [yourlabs/django-autocomplete-light#268]
The use of `User.objects` in `autocomplete_light/tests/registry.py` produces the exception below if you try to run "manage.py test" in a project that has a swapped out User model:

```
Traceback (most recent call last):
  File "./manage.py", line 8, in <module>
    management.execute_from_command_line()
  File "/home/luke/.virtualenvs/wolfandbadger/local/lib/python2.7/site-packages/django/core/management/__init__.py", line 453, in execute_from_command_line
    utility.execute()
  File "/home/luke/.virtualenvs/wolfandbadger/local/lib/python2.7/site-packages/django/core/management/__init__.py", line 392, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/home/luke/.virtualenvs/wolfandbadger/local/lib/python2.7/site-packages/django/core/management/commands/test.py", line 51, in run_from_argv
    super(Command, self).run_from_argv(argv)
  File "/home/luke/.virtualenvs/wolfandbadger/local/lib/python2.7/site-packages/django/core/management/base.py", line 222, in run_from_argv
    self.execute(*args, **options.__dict__)
  File "/home/luke/.virtualenvs/wolfandbadger/local/lib/python2.7/site-packages/django/core/management/commands/test.py", line 72, in execute
    super(Command, self).execute(*args, **options)
  File "/home/luke/.virtualenvs/wolfandbadger/local/lib/python2.7/site-packages/django/core/management/base.py", line 255, in execute
    output = self.handle(*args, **options)
  File "/home/luke/.virtualenvs/wolfandbadger/local/lib/python2.7/site-packages/django/core/management/commands/test.py", line 89, in handle
    failures = test_runner.run_tests(test_labels)
  File "/home/luke/.virtualenvs/wolfandbadger/local/lib/python2.7/site-packages/django/test/simple.py", line 369, in run_tests
    suite = self.build_suite(test_labels, extra_tests)
  File "/home/luke/.virtualenvs/wolfandbadger/local/lib/python2.7/site-packages/django/test/simple.py", line 260, in build_suite
    suite.addTest(build_suite(app))
  File "/home/luke/.virtualenvs/wolfandbadger/local/lib/python2.7/site-packages/django/test/simple.py", line 68, in build_suite
    test_module = get_tests(app_module)
  File "/home/luke/.virtualenvs/wolfandbadger/local/lib/python2.7/site-packages/django/test/simple.py", line 25, in get_tests
    test_module = import_module('.'.join(prefix + [TEST_MODULE]))
  File "/home/luke/.virtualenvs/wolfandbadger/local/lib/python2.7/site-packages/django/utils/importlib.py", line 35, in import_module
    __import__(name)
  File "/home/luke/.virtualenvs/wolfandbadger/local/lib/python2.7/site-packages/autocomplete_light/tests/__init__.py", line 16, in <module>
    from .registry import (
  File "/home/luke/.virtualenvs/wolfandbadger/local/lib/python2.7/site-packages/autocomplete_light/tests/registry.py", line 21, in <module>
    class Generic(autocomplete_light.AutocompleteGenericBase):
  File "/home/luke/.virtualenvs/wolfandbadger/local/lib/python2.7/site-packages/autocomplete_light/tests/registry.py", line 23, in Generic
    User.objects.all(),
  File "/home/luke/.virtualenvs/wolfandbadger/local/lib/python2.7/site-packages/django/db/models/manager.py", line 256, in __get__
    self.model._meta.object_name, self.model._meta.swapped
AttributeError: Manager isn't available; User has been swapped for 'accounts.User'
```

(Actually, getting this traceback is quite tricky, because "manage.py test" swallows all but the last line, making it rather tricky to diagnose until you modify Django source).

Because this error happens at import time (when the 'Generic' class is defined), it stops test discovery dead, rather than just resulting in failing tests, which is kind of annoying.

There are probably other problems with the test suite for this scenario, this is just the first one I hit.

**Comment by jpic** [comment]
Django doesn't make much difference between functional and unit tests: `./manage.py test` will run all your tests.

It would be nice to not enable integration tests by default.

**Comment by jpic** [comment]
I'll try for 2.3, but can't garantee.

**Comment by jpic** [comment]
I'm a bit confused about this issue, should we move the tests outside the package ie. in repo/tests, or should we make them pass when run from user-projects ?

What's the best practice here ?

**Comment by jpic** [comment]
Probably has been fixed in 3.0, otherwise please reopen.
